### PR TITLE
Issue #3037: DATE + TIME

### PR DESCRIPTION
**Pull request by hawkfish**
Implement and test missing function.

### CODE REVIEW

File: src/function/scalar/operators/add.cpp [hunk 1]
```diff
@@ -56,6 +56,20 @@ date_t AddOperator::Operation(int32_t left, date_t right) {
 	return AddOperator::Operation<date_t, int32_t, date_t>(right, left);
 }
 
+template <>
+timestamp_t AddOperator::Operation(date_t left, dtime_t right) {
+	timestamp_t result;
+	if (!Timestamp::TryFromDatetime(left, right, result)) {
+		throw OutOfRangeException("Timestamp out of range");
```
Comment (Mytherin) [inline src/function/scalar/operators/add.cpp:63]: Could we add a test that covers this line?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks great. Minor comment:

File: src/function/scalar/operators/add.cpp (hunk 1 above)
Comment (hawkfish) [reply src/function/scalar/operators/add.cpp:63]: Incoming.

**Review by Mytherin — APPROVED** [review]
Ready for merge after feature freeze ends.

**Pull request merged**